emop-maven-plugin/src/main/java/edu/cornell/VmsMojo.java

```diff
@@ -0,0 +1,277 @@
+                    // Gets renamed classes
+                    if (!diff.getOldPath().equals(diff.getNewPath())) {
+                        renames.put(diff.getNewPath(), diff.getOldPath());
+                    }
+
+                    // For each file, find the replacements, additions, and deletions at each line
+                    for (Edit edit : diffFormatter.toFileHeader(diff).toEditList()) {
+                        Map<Integer, Integer> lineChange = new HashMap<>();
+                        if (lineChanges.containsKey(diff.getOldPath())) {
+                            lineChange = lineChanges.get(diff.getOldPath());
+                        }
+                        lineChange.put(edit.getBeginA() + 1, edit.getLengthB() - edit.getLengthA());
+                        lineChanges.put(diff.getOldPath(), lineChange);
+                    }
+                }
+            }
+        } catch (IOException exception) {
+            throw new MojoExecutionException("Encountered an error when comparing different files");
+        }
+    }
+
+    /**
+     * Scrubs newViolations of violations believed to be duplicates from violation-counts-old.
+     */
+    private void removeDuplicateViolations() {
+        Set<Violation> violationsToRemove = new HashSet<>();
+        for (Violation newViolation : newViolations) {
+            Set<Violation> relevantOldViolations = oldViolations.stream()
+                    .filter(oldViolation -> oldViolation.getSpecification().equals(newViolation.getSpecification()))
+                    .filter(oldViolation -> oldViolation.getClassInfo().equals(newViolation.getClassInfo())
+                                         || isRenamed(oldViolation.getClassInfo(), newViolation.getClassInfo()))
+                    .filter(oldViolation -> hasSameLineNumber(oldViolation.getClassInfo(), oldViolation.getLineNum(),
+                            newViolation.getLineNum()))
+                    .collect(Collectors.toSet());
+
+            if (!relevantOldViolations.isEmpty()) {
+                violationsToRemove.add(newViolation);
+            }
+        }
+        for (Violation violationToRemove : violationsToRemove) {
+            newViolations.remove(violationToRemove);
+        }
+    }
+
+    /**
+     * Determines whether an old class has been renamed to the new one or not.
+     *
+     * @param oldClass Previous possible name of a class
+     * @param newClass Rename being considered
+     * @return Whether the old class name was renamed to the new one
+     */
+    private boolean isRenamed(String oldClass, String newClass) {
+        for (String renamedClass : renames.keySet()) {
+            if (renamedClass.contains(newClass) && renames.get(renamedClass).contains(oldClass)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Determines whether an old line in a file can be mapped to the new line. There's room for optimization.
```

Comment:
> There's room for optimization.

Make this into a TODO comment and say how you are thinking to do the optimization